tsdb/meta.go

```diff
@@ -598,7 +598,9 @@ func (m *Measurement) AppendSeriesKeysByID(dst []string, ids []uint64) []string
 	m.mu.RLock()
 	defer m.mu.RUnlock()
 	for _, id := range ids {
-		dst = append(dst, m.seriesByID[id].Key)
+		if s, ok := m.seriesByID[id]; ok && s != nil {
```

Comment:
True.  I'm not positive somewhere else in the code is not setting or will set `s.seriesByID[id] = nil` though.